tests/test-cli.js

```diff
@@ -0,0 +1,760 @@
+/* eslint-disable wrap-regex */
+"use strict"
+
+const loadFromDeps = (typeof process !== "undefined" && process.argv.length >= 2 && process.argv[1].match(/ospec\/node_modules\/\.bin\/ospec$/))
+const o = loadFromDeps ? require("ospec") : require("../ospec")
+const fs = require("fs")
+const {copyFile, lstat, mkdir, readdir, rmdir, symlink, unlink, writeFile} = fs.promises
+const {join} = require("path")
+const chp = require("child_process")
+
+const projectCwd = process.cwd()
+const ospecPkgJsonPath = join(projectCwd, "package.json")
+const ospecLibPath = join(projectCwd, "ospec.js")
+const ospecBinPath = join(projectCwd, "bin/ospec")
+const fixturesDir = join(projectCwd, "./tests/fixtures")
+
+function stringify(arg) {
+	return JSON.stringify(arg, null, 2)
+}
+
+const commands = ["node", "npm", "yarn"].filter((launcher) => {
+	try {
+		chp.execFileSync(launcher, ["-v"])
+		return true
+	} catch(e) {
+		return false
+	}
+})
+
+// not sure how that could happen...
+if (commands.length === 0) throw new Error("couldn't find either node, npm nor yarn")
+
+const isWindows = process.platform === "win32"
+
+const winBashscript = `#!/bin/sh
+basedir=$(dirname "$(echo "$0" | sed -e 's,\\,/,g')")
+
+case \`uname\` in
+    *CYGWIN*) basedir=\`cygpath -w "$basedir"\`;;
+esac
+
+node  "$basedir/../mime/cli.js" "$@"
+ret=$?
+
+exit $ret
+`
+
+const winCmdScript = `
+@SETLOCAL
+@SET PATHEXT=%PATHEXT:;.JS;=;%
+node  "%~dp0\\..\\ospec\\bin\\ospec" %*	
+`
+
+const finalizeBinaryInstall = isWindows
+	? async (path) => {
+		await writeFile(join(path, "ospec.cmd"), winCmdScript)
+		await writeFile(join(path, "ospec"), winBashscript)
+	}
+	: async (path) => {
+		await symlink("../ospec/bin/ospec", join(path, "ospec"))
+	}
+
+function execFile(command, args, options) {
+	if (typeof options.cwd !== "string") throw new Error(`\`options.cwd\`: string expected for ${command} ${stringify(args)} ${stringify(options)}`)
+	return new Promise((fulfill, reject) => {
+		chp.execFile(command, args, {
+			shell: true,
+			...options,
+			timeout: 5000,
+			killSignal: "SIGKILL",
+			windowsHide: true
+		}, (error, stdout, stderr) => {
+			if (error != null && typeof error.code !== "number") {
+				reject(error)
+			} else {
+				fulfill({
+					code: error == null ? 0 : error.code,
+					stdout,
+					stderr
+				})
+			}
+		})
+	})
+}
+
+function removeWrarnings(stderr) {
```

Comment:
`s/removeWrarnings/removeWarnings/g`

You also replicated this typo in every one of the 19 uses in this file. 😄 (Yes, I did a Ctrl+F.)

Reply:
Auto-completion FTW!